lib/extractor/docker-archive/index.ts

```diff
@@ -34,3 +38,32 @@ export function getPlatformFromConfig(
     ? `${imageConfig.os}/${imageConfig.architecture}`
     : undefined;
 }
+
+export function getDetectedLayersInfoFromConfig(imageConfig) {
+  const runInstructions = getUserInstructionLayersFromConfig(imageConfig)
+    .filter((instruction) => !instruction.empty_layer)
+    .map((instruction) => instruction.created_by);
+
+  const packages = getPackagesFromRunInstructions(runInstructions);
+  const layers = getLayersFromPackages(packages);
+  return { packages, layers };
+}
+
+export function getUserInstructionLayersFromConfig(imageConfig) {
+  const diffInHours = (d1, d2) => Math.abs(d1 - d2) / 1000 / (60 * 60);
```

Comment:
Have we looked at using Luxon for time/date calculations? There are a lot of hairy edge cases around time and date that it handles, plus it gives a nice readable API.

Reply:
Personally I feel that a lib for this one line is a bit much, especially since it's not too important that it will be right on the second (there's a lot of buffer time)
WDYT?

Reply:
Don't have a strong opinion for this use case